IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m

```diff
@@ -40,62 +72,85 @@ @interface MSIDLastRequestTelemetry()
 
 @implementation MSIDLastRequestTelemetry
 
+static bool shouldReadFromDisk = YES;
+static const NSInteger currentSchemaVersion = 2;
+
 #pragma mark - Init
 
-- (id)initInternal
+- (instancetype)initInternal
 {
     self = [super init];
     if (self)
     {
-        _schemaVersion = 2;
-        
-        NSString *queueName = [NSString stringWithFormat:@"com.microsoft.msidlastrequesttelemetry-%@", [NSUUID UUID].UUIDString];
-        _synchronizationQueue = dispatch_queue_create([queueName cStringUsingEncoding:NSASCIIStringEncoding], DISPATCH_QUEUE_CONCURRENT);
+        _schemaVersion = currentSchemaVersion;
+        _synchronizationQueue = [self initializeDispatchQueue];
     }
     return self;
 }
 
+- (instancetype)initFromDisk
+{
+    NSString *saveLocation = [self filePathToSavedTelemetry];
+    if (saveLocation && [[NSFileManager defaultManager] fileExistsAtPath:saveLocation])
+    {
+        NSData *dataToUnarchive = [NSData dataWithContentsOfFile:saveLocation];
+        NSKeyedUnarchiver *unarchiver = [NSKeyedUnarchiver msidCreateForReadingFromData:dataToUnarchive error:nil];
```

Comment:
nit: might be worth logging an error if we get corrupt data

Reply:
Thanks, addressed